pkg/network/cache/multi.go

```diff
@@ -186,6 +187,11 @@ func (x *multiClient) ReportError(err error) {
 		return
 	}
 
+	// NeoFS status responses mean connection is OK
+	if errors.Is(err, apistatus.Error) {
```

Comment:
What if it's `ErrWrongMagicNumber`? `ErrSignatureVerification`? `ErrNodeUnderMaintenance`? Some other errors can be expected or not depending on request.